src/Chapter2.hs

```diff
@@ -348,8 +352,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf ls = fst . splitAt half $ ls
+  where
+    half = length ls `div` 2
```

Comment:
This is a correct solution! Though, `splitAt` does more than required. Here, it's enough to _take_ the first half of the list